spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -563,4 +635,58 @@ private void unlockNotify(String lockKey) {
 
 	}
 
+	private final class RedisSpinLock extends RedisLock {
+		private static final String OBTAIN_LOCK_SCRIPT =
+				"local lockClientId = redis.call('GET', KEYS[1])\n" +
+				"if lockClientId == ARGV[1] then\n" +
+				"  redis.call('PEXPIRE', KEYS[1], ARGV[2])\n" +
+				"  return true\n" +
+				"elseif not lockClientId then\n" +
+				"  redis.call('SET', KEYS[1], ARGV[1], 'PX', ARGV[2])\n" +
+				"  return true\n" +
+				"end\n" +
+				"return false";
+		private final RedisScript<Boolean> obtainLockScript = new DefaultRedisScript<>(OBTAIN_LOCK_SCRIPT, Boolean.class);
+
+		private RedisSpinLock(String path) {
+			super(path);
+		}
+
+		@Override
+		protected boolean tryRedisLockInner(long time) throws ExecutionException, InterruptedException {
+			long now = System.currentTimeMillis();
+			if (time == -1L) {
+				while (!obtainLock()) {
+					Thread.sleep(100); //NOSONAR
+				}
+				return true;
+			}
+			else {
+				long expire = now + TimeUnit.MILLISECONDS.convert(time, TimeUnit.MILLISECONDS);
+				boolean acquired;
+				while (!(acquired = obtainLock()) && System.currentTimeMillis() < expire) { //NOSONAR
+					Thread.sleep(100); //NOSONAR
+				}
+				return acquired;
+			}
+		}
+
+		@Override
+		protected void removeLockKeyInnerUnlink() {
+			RedisLockRegistry.this.redisTemplate.unlink(this.lockKey);
+		}
+
+		@Override
+		protected void removeLockKeyInnerDelete() {
+			RedisLockRegistry.this.redisTemplate.delete(this.lockKey);
+		}
+
+		private boolean obtainLock() {
```

Comment:
Isn't this method the same as in the `RedisPubSubLock`?
Cannot it go to the super class, too?